app/src/main/java/com/tohami/newsapi/di/Injection.java

```diff
@@ -0,0 +1,25 @@
+package com.tohami.newsapi.di;
+
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+
+import com.tohami.newsapi.data.retrofit.ServiceHelper;
+import com.tohami.newsapi.factory.RepositoryFactory;
+import com.tohami.newsapi.factory.ViewModelFactory;
+
+public class Injection {
+
+
+    @NonNull
+    private static RepositoryFactory provideRepositoryFactory(Context context) {
+        return new RepositoryFactory(ServiceHelper.getInstance().getRetrofitInterface(context));
+    }
+
+
+    @NonNull
+    public static ViewModelFactory provideViewModelFactory(Context context) {
```

Comment:
each view model should have its own factory

Reply:
done